src/libraries/Microsoft.PowerFx.Core/Syntax/UserDefinitions.cs

```diff
@@ -99,18 +99,20 @@ private IEnumerable<UserDefinedFunction> CreateUserDefinedFunctions(IEnumerable<
 
             foreach (var udf in uDFs)
             {
-                CheckParameters(udf.Args, errors, out var paramsRecordType);
-                CheckReturnType(udf.ReturnType, errors);
-
                 var udfName = udf.Ident.Name;
-                var func = new UserDefinedFunction(udfName.Value, udf.ReturnType, udf.Body, udf.IsImperative, udf.Args, paramsRecordType);
-
                 if (texlFunctionSet.AnyWithName(udfName) || BuiltinFunctionsCore._library.AnyWithName(udfName))
                 {
                     errors.Add(new TexlError(udf.Ident, DocumentErrorSeverity.Severe, TexlStrings.ErrUDF_FunctionAlreadyDefined, udfName));
                     continue;
                 }
 
+                if (!(CheckParameters(udf.Args, errors) & CheckReturnType(udf.ReturnType, errors)))
```

Comment:
'&' to evaluate both CheckParameters() and CheckReturnType() so we can add errors on the nodes